commands/curl_put.go

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package commands
+
+import (
+	"github.com/spf13/cobra"
+)
+
+const curlPutCommandName = "put"
+
+var curlPutExample = `
+# Create a knn index from mapping setting saved in file "knn-mapping.json"
+odfe-cli curl put --path "my-knn-index"  \
+                  --data "@some-location/knn-mapping.json" \
```

Comment:
Out of curiousity: is prefixing file input with @ a convention in another cli?

Reply:
Not really, usually there will be two different flags. This i borrowed from curl. Do you prefer as another flag "--data-file-path" something like that?

Reply:
If this is how curl does it, thats fine